backend/controller/timeline/events_ingress.go

```diff
@@ -3,31 +3,141 @@ package timeline
 import (
 	"context"
 	"encoding/json"
+	"io"
+	"net/http"
+	"time"
 
 	"github.com/alecthomas/types/optional"
 
+	ftlencryption "github.com/TBD54566975/ftl/backend/controller/encryption/api"
 	"github.com/TBD54566975/ftl/backend/controller/timeline/internal/sql"
 	"github.com/TBD54566975/ftl/backend/libdal"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
 	"github.com/TBD54566975/ftl/internal/model"
 )
 
-type IngresEvent struct {
+type IngressEvent struct {
+	ID            int64
 	DeploymentKey model.DeploymentKey
-	RequestKey    model.RequestKey
+	RequestKey    optional.Option[model.RequestKey]
+	Verb          schema.Ref
+	Method        string
+	Path          string
+
+	StatusCode     int
+	Time           time.Time
+	Duration       time.Duration
+	Request        json.RawMessage
+	RequestHeader  json.RawMessage
+	Response       json.RawMessage
+	ResponseHeader json.RawMessage
+	Error          optional.Option[string]
 }
 
-// The internal JSON payload of an ingress event.
+func (e *IngressEvent) GetID() int64 { return e.ID }
+func (e *IngressEvent) event()       {}
+
 type eventIngressJSON struct {
-	DurationMS int64                   `json:"duration_ms"`
-	Request    json.RawMessage         `json:"request"`
-	Response   json.RawMessage         `json:"response"`
-	Error      optional.Option[string] `json:"error,omitempty"`
-	Stack      optional.Option[string] `json:"stack,omitempty"`
+	DurationMS     int64                   `json:"duration_ms"`
+	Method         string                  `json:"method"`
+	Path           string                  `json:"path"`
+	StatusCode     int                     `json:"status_code"`
+	Request        json.RawMessage         `json:"request"`
+	RequestHeader  json.RawMessage         `json:"req_header"`
+	Response       json.RawMessage         `json:"response"`
+	ResponseHeader json.RawMessage         `json:"res_header"`
+	Error          optional.Option[string] `json:"error,omitempty"`
+}
+
+type Ingress struct {
```

Comment:
Wanna nest this in `IngressEvent`?

Reply:
I very much do, but they are a bit too different. Like when creating an event I never want the requestKey to be missing, but it's allowed to be in the db. Similar to `Request` and `Response` where since they are different types it might get confusing to users of the API. 

I had to do something similar with `CallEvent` :(

Reply:
ahhhh got it, that makes sense